README.md

```diff
@@ -1,9 +1,243 @@
 # ros_utilities
 Wrappers and other utilities for ROS2
 
+## Getting started
+
+### Process-wide APIs
+
+These APIs wrap [`rclpy`](https://github.com/ros2/rclpy) APIs to provide a simpler UX in common use cases.
+All APIs are thread-safe and exception-safe. Resource management is automatic. By default, callbacks are
+dispatched in the background, enabling both synchronous and asynchronous programming.
+
+These APIs are also quite handy to reconcile past [`rospy`](http://wiki.ros.org/rospy) experience with ROS 2.
+
+#### Setting up single node processes
+
+Just decorate your executable entrypoint (or `main` function) with `bdai_ros2_wrappers.process.main`:
+
+```python
+import logging
+import time
+
+import bdai_ros2_wrappers.process as ros_process
+
+import std_msgs.msg
+
+@ros_process.main()
+def entrypoint() -> None:
+    # no need to initialize, it is automatic
+    node = ros_process.node()  # or entrypoint.node
+    assert node is not None
+    pub = node.create_publisher(std_msgs.msg.String, "test", 1)
+
+    def callback() -> None:
+        time.sleep(10)  # you can block in a callback
+        pub.publish(std_msgs.msg.String(data="testing"))
+    executor = ros_process.executor()  # or entrypoint.executor
+    assert executor is not None
+    executor.create_task(callback)
+
+    time.sleep(10)  # you can block in the main thread
+
+    node.get_logger().info("testing")
+    logging.info("testing")  # you can use Python logging
+
+    return  # no need to cleanup or shutdown, it is automatic
+
+if __name__ == "__main__":
+    entrypoint()
+```
+
+This is ideal for quick prototyping and simple scripts.
+
+#### Setting up multi-node processes
+
+You can spin as many ROS 2 nodes as you need, and skip the default process-wide node if unnecessary:
+
+```python
+import logging
+import time
+
+from typing import Any, List
+
+import bdai_ros2_wrappers.process as ros_process
+from bdai_ros2_wrappers.node import Node
+
+def graph(**kwargs: Any) -> List[Node]:
+    # make sure to forward all keyword arguments for proper loading!
+    return [Node("my_node", **kwargs), Node("my_other_node", **kwargs)]
+
+@ros_process.main(prebaked=False)
+def main() -> None:
+    ros_process.load(Node, "_hidden_node")  # or main.load
```

Comment:
Is this to show that we can throw in another node into the processes and that when we call spin(graph) will that then spin all 3 nodes?

Reply:
Exactly. And that can be done from anywhere below `main` in the call stack.

Reply:
It will use defaults, but you can configure it before that happens:

```python
from bdai_ros2_wrappers.executors import AutoScalingMultiThreadedExecutor
from bdai_ros2_wrappers.executors import foreground

# ...

@ros_process.main(prebaked=False)
def main() -> None:
    with foreground(AutoScalingMultiThreadedExecutor(max_threads=100)) as main.executor:
        ros_process.load(Node, "_hidden_node")  # or main.load
        ros_process.spin(graph)  # or main.spin
```

Reply:
Awesome!

Reply:
Can the `README` be updated to explain that this will spin all three nodes and perhaps address a little more what `Load` does?  I had the same question reading this as @mpickett-bdai

Reply:
Done in https://github.com/bdaiinstitute/ros_utilities/pull/32/commits/efb0db0d1ef57bdecc49aaff0dcbe9222defb1e3.